src/Raven.Server/NotificationCenter/SlowWriteNotifications.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            _notificationCenter = notificationCenter;
+            _notificationsStorage = notificationsStorage;
+            _database = database;
+            _slowWrites = new ConcurrentDictionary<string, SlowWritesDetails.SlowWriteInfo>();
+        }
+
+        public void Add(string path, double dataSizeInMb, double durationInSec)
+        {
+            var now = SystemTime.UtcNow;
+
+            if (_slowWrites.TryGetValue(path, out var info))
+            {
+                if (now - info.Date < UpdateFrequency)
+                    return;
+
+                info.DataWrittenInMb = dataSizeInMb;
+                info.DurationInSec = durationInSec;
+                info.SpeedInMbPerSec = dataSizeInMb / durationInSec;
+                info.Date = now;
+            }
+            else
+            {
+                info = new SlowWritesDetails.SlowWriteInfo
+                {
+                    Path = path,
+                    DataWrittenInMb = dataSizeInMb,
+                    DurationInSec = durationInSec,
+                    SpeedInMbPerSec = dataSizeInMb / durationInSec,
+                    Date = now
+                };
+
+                _slowWrites.TryAdd(path, info);
+            }
+
+            _updateNotificationInStorageRequired = true;
+
+            if (_timer != null) 
+                return;
+
+            lock (_pagerCreationLock)
+            {
+                if (_timer != null) 
+                    return;
+
+                _timer = new Timer(UpdateNotificationInStorage, null, TimeSpan.FromSeconds(10), TimeSpan.FromMinutes(1));
+            }
+        }
+
+        internal void UpdateNotificationInStorage(object state)
+        {
+            if (_updateNotificationInStorageRequired == false)
+                return;
+
+            var hint = GetOrCreateSlowWrites(out var details);
+
+            foreach (var info in _slowWrites)
+            {
+                details.Writes[info.Key] = info.Value;
+            }
+
+            _notificationCenter.Add(hint);
+
+            _updateNotificationInStorageRequired = false;
+        }
+
+        internal SlowWritesDetails GetSlowWritesDetails()
+        {
+            GetOrCreateSlowWrites(out var details);
+            return details;
+        }
+
+        private PerformanceHint GetOrCreateSlowWrites(out SlowWritesDetails details)
+        {
+            const string source = "slow-writes";
+
+            var id = PerformanceHint.GetKey(PerformanceHintType.SlowIO, source);
+
+            using (_notificationsStorage.Read(id, out var ntv))
+            {
+                if (ntv == null || ntv.Json.TryGet(nameof(PerformanceHint.Details), out BlittableJsonReaderObject detailsJson) == false || detailsJson == null)
+                {
+                    details = new SlowWritesDetails();
+                }
+                else
+                {
+                    details = (SlowWritesDetails)EntityToBlittable.ConvertToEntity(
```

Comment:
Not a fan of using EntityToBlittable on server side, in that case it wont be too frequent though..

Reply:
It's easy to use. Is there any problem with that? We use it quite commonly when dealing with aggregated notifications (as you pointed, those aren't frequently running actions).

Reply:
As long as it is not frequently called, it's bearable, but in general, I would feel more comfortable if we would use our "compiled" deserialization, which is more performant.

Reply:
Oh, yeah but it's not so important in those scenarios